bes_publisher/main.go

```diff
@@ -21,15 +25,40 @@ var (
 		50*1024*1024,
 		"Maximum receive message size in bytes accepted by gRPC methods",
 	)
+	gcpProjectID = flag.String(
+		"gcp_project_id",
+		"",
+		"GCP project with PubSub resources to use",
+	)
+	besPubsubTopic = flag.String(
+		"bes_pubsub_topic",
+		"",
+		"Name of topic to publish BES messages on",
+	)
 )
 
+func exitIf(err error) {
+	if err != nil {
+		glog.Exit(err)
+	}
+}
+
 func main() {
 	flag.Parse()
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
```

Comment:
This causes the app to hang and not get killed. To handle this, wait on the ctx.Done() channel and kill the app when received.

Reply:
Fixed in next PR